### ISSUE TITLE
Strange OSPF routes

### ISSUE THREAD

**Issue by ManiAm** [inet-framework/inet#367]
While I was working on an OSPF example, I noticed an strange behavior by OSPF. You can find this example in Step5 scenario in here:

https://github.com/ManiAm/inet/blob/INETex/examples_INETex/static_dynamic_routing/omnetpp.ini

So here is what happens:

![2018-07-23_14-11-21](https://user-images.githubusercontent.com/5473998/43103875-5807db40-8e84-11e8-8a76-4b58492f4f05.png)

Look at the routing table of router0. OSPF has added a remote network (first entry) for 10.0.0.17/24 that is the directly-connected interface of router0. And look at the route :) It says if I receive any datagram with destination 10.0.0.17, then forward it to eth2. The datagram takes the route I showed with a red arrow. It goes around the inter-network and goes back to router0. The cost of this path, as calculated correctly, is 13 (I have shown the output cost of each interface along the path and total cost is 1+ 1+ 1 + 10 = 13). The correct route, however, is the one added by IFACENETMASK (as I highlighted with a red rectangle). So my question is why OSPF has added this route?

**Comment by levy** [comment]
Thanks for the thorough description and explanation! We are going to review this with Andras who knows more about OSPF than me.

**Comment by ManiAm** [comment]
I think I found the cause of this problem. After checking the following method that is responsible for calculating the SPF tree, I noticed that the algorithm works like a charm and the problem is in the `newRoutingTable` parameter that is passed into this method.

https://github.com/inet-framework/inet/blob/163f2de0de5f19d190240125c2dee15c81d32c85/src/inet/routing/ospfv2/router/OspfArea.cc#L1396

`newRoutingTable` is empty and OSPF assumes that the routing table of the router is empty. That's why it tries to add those strange entries into the routing table. If OSPF knew that the routing table is already populated with directly-connected entries, it will never add those strange entries, because the cost of those routes are higher than directly-connected routes. The solution is very simple. We need to add the directly-connected routes into `newRoutingTable` before passing it to the method. I fixed this in commit 1992ac9. The scenario above works perfectly now and the routing table of router0, for example, looks like this (pheeeww).

![2018-07-26_23-04-27](https://user-images.githubusercontent.com/5473998/43304841-6f1893b2-912a-11e8-8a44-4fced19a73ec.png)

I am closing this issue.
